pkg/processor/worker/worker.go

```diff
@@ -160,25 +156,19 @@ func (w *Worker) Terminate() error {
 }
 
 func (w *Worker) Drain() error {
-	w.drainedLock.Lock()
-	defer w.drainedLock.Unlock()
-
-	if !w.isDrained.Load() {
-		err := w.runtime.Drain()
-		if err == nil {
-			w.logger.DebugWith("Successfully drained worker", "workerIndex", w.index)
-			w.isDrained.Store(true)
-		}
+	if err := w.runtime.Drain(); err != nil {
```

Comment:
I know we discussed this, but removing the `isDrained` check will cause MANY signal being sent from the processor to the python process.
E.g in the case of N partitions and 1 workers, now we get N syscalls to the python wrapper.

While it will indeed be handled in the wrapper by discarding the events, I am not 100% sure this is the best idea...

Reply:
@TomerShor On one hand, we have enhanced efficiency on the processor side by eliminating unnecessary mutexes. On the other hand, we can now send multiple `drain` signals simultaneously. However, based on my understanding, this should not be a significant issue, as in the draining handler on the wrapper side, we simply set the draining flag to true.

Also, another thing here to discuss is what if we got draining when `discard_events==True`? I was thinking if we can skip draining in that case because if it's set to true, it means that we have already performed draining.

Reply:
I agree that it should be ignored if `discard_events` is set to True